mlir/tools/mlir-miopen-lib/mlir-miopen-lib.hpp

```diff
@@ -1,6 +1,6 @@
 #include <cstddef>
 
-namespace mlir {
+#define MLIRMIOPEN_VERSION_FLAT 0
```

Comment:
Please return namespace. It is bad practice to export C++ API in a global namespace. It is much better to remove "Mlir" from function names and types (but not from the macros).

Reply:
I believe for extend-ability, our API should be pure C API, not C++ API.

The namespace and extern C are conflicting message. extern C disallows name mangling, on the other hand, namespace will force mangled name again.

Reply:
This file does not look as a C header. The C header file should not `#include <cstddef>`. Filename extension should not be `.hpp`.

Reply:
Addressed in https://github.com/ROCmSoftwarePlatform/llvm-project-mlir/pull/64